src/conn_handler.c

```diff
@@ -227,9 +228,11 @@ static int stream_formatter_bin(FILE *pipe, void *data, metric_type type, char *
             STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_STDDEV, timer_stddev(&t->tm));
             STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_MIN, timer_min(&t->tm));
             STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_MAX, timer_max(&t->tm));
-            STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT | 50, timer_query(&t->tm, 0.5));
-            STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT | 95, timer_query(&t->tm, 0.95));
-            STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT | 99, timer_query(&t->tm, 0.99));
+            for (i=0; i < GLOBAL_CONFIG->num_quantiles; i++) {
+                STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT |
+                    (int)(GLOBAL_CONFIG->quantiles[i] * 100),
```

Comment:
This may get to be a bit screwy given the lack of precision for things like 99.9%